### PR TITLE
Improved workflow

### PR DESCRIPTION
**Pull request by mr-bjerre**
- [x] add [`poetry`](https://python-poetry.org/) for package and dependecy management
- [x] add coverage: [`pytest-cov`](https://github.com/pytest-dev/pytest-cov)
- [x] add formatters: [`black`](https://github.com/psf/black) and [`isort`](https://pycqa.github.io/isort/)
- [x] add linter: [`flake8`](https://pypi.org/project/flake8/) and/or [`pylint`](https://www.pylint.org/)
- [x] add static type checking: [`mypy`](http://mypy-lang.org/) and/or [`pyright`](https://github.com/microsoft/pyright)
- [x] add GitHub actions, e.g. [CodeCoverage](https://github.com/marketplace/codecov)
- [ ] add initial docs, e.g. using [sphinx](https://www.sphinx-doc.org/en/master/)

### CODE REVIEW

**Comment by mr-bjerre** [comment]
@davidkpiano I'm not sure if you want all of the above in a single PR where each bullet point is its own commit (did the first two in a single commit though)? 

I'll await your comments before proceeding.

**Comment by davidkpiano** [comment]
I think we can just do it in one PR.

**Comment by mr-bjerre** [comment]
Cool I'll keep checking off the checkboxes in this PR then

**Comment by mr-bjerre** [comment]
There are quite a lot of type errors. I suggest we fix that in another PR - I've just added the type checker to the environment for now.

@davidkpiano can you navigate to http://codecov.io/, and log in with your GitHub account and find your repository? 

I've added an action that should run tests and code quality checks on PRs (requires CodeCov for reporting in GitHub). I've also added a release action that requires a github secret.

**Comment by mr-bjerre** [comment]
I'm testing the actions here btw: https://github.com/mr-bjerre/xstate-python/pull/1

Not sure how else to do it since I probably can't enable actions in your repo.

File: examples/actions.py [hunk 1]
```diff
@@ -1,54 +1,55 @@
...
-                "exit": [{"type": "exitGreen"}]
-            },
-            "yellow": {
-                "on": {"TIMER": "red"}, 
-                "entry": [{"type": "enterYellow"}]
+                "on": {"TIMER": "yellow"},
+                "entry": [{"type": "enterGreen"}],
+                "exit": [{"type": "exitGreen"}],
             },
+            "yellow": {"on": {"TIMER": "red"}, "entry": [{"type": "enterYellow"}]},
```
Comment (davidkpiano) [inline examples/actions.py:30]: Is this the autoformatter's doing? Wonder if we can force newlines

**Review by davidkpiano — APPROVED** [review]

File: examples/actions.py (hunk 1 above)
Comment (mr-bjerre) [reply examples/actions.py:30]: ah yes it is. With `black` you can do

```python
# fmt: off
<UNFORMATTED CODE>
# fmt: on
```

Probably good to do that in the examples

**Comment by mr-bjerre** [comment]
Closes #33

Created new issue to resolve all type issues: #35 
Created new issue to add the documentation in another PR: #36

**Pull request merged**